### ISSUE TITLE
Proxy by patterns not recognized by default

### ISSUE THREAD

**Issue by dromhost** [foxyproxy/browser-extension#164]
Firefox 133.0 (64-bit) Linux
FoxyProxy 8.9

When Firefox is restarted, FoxyProxy is set to "Proxy by Patterns", although this does not happen. Patterns are not working.
If I click "Disable", and then "Proxy by Patterns" again, things start working by the added patterns. If I forget when restarting Firefox, security is compromized since proxies are not used at all.

Please advise

**Comment by erosman** [comment]
Can you paste the exported settings of FoxyProxy?
(remove private data before pasting)

**Comment by dromhost** [comment]
Thanks @erosman 
This is the exported .json with some data removed.

`{
  "mode": "pattern",
  "sync": true,
  "autoBackup": false,
  "passthrough": "",
  "theme": "",
  "container": {
    "incognito": "",
    "container-1": "",
    "container-2": "",
    "container-3": "",
    "container-4": ""
  },
  "commands": {
    "setProxy": "",
    "setTabProxy": "",
    "quickAdd": ""
  },
  "data": [
    {
      "active": true,
      "title": "ssh35393",
      "type": "socks5",
      "hostname": "127.0.0.1",
      "port": "35393",
      "username": "",
      "password": "",
      "cc": "",
      "city": "",
      "color": "#cc1cc9",
      "pac": "",
      "pacString": "",
      "proxyDNS": true,
      "include": [
        {
          "type": "wildcard",
          "title": "*binsearch*",
          "pattern": "*://*binsearch*/",
          "active": true
        },
        {
          "type": "wildcard",
          "title": "*whatismyip*",
          "pattern": "*whatismyip*",
          "active": true
        },
        {
          "type": "wildcard",
          "title": "*duckduckgo*",
          "pattern": "*duckduckgo*",
          "active": true
        },
        {
          "type": "wildcard",
          "title": "*canyouseeme*",
          "pattern": "*canyouseeme*",
          "active": true
        },
      ],
      "exclude": [
        {
          "type": "regex",
          "title": "local hostnames (usually no dots in the name). Pattern exists because 'Do not use this proxy for localhost and intranet/private IP addresses' is checked.",
          "pattern": "^[^:]+://(?:[^:@/]+(?::[^@/]+)?@)?(?:localhost|127\\.\\d+\\.\\d+\\.\\d+)(?::\\d+)?(?:/.*)?/",
          "active": true
        },
        {
          "type": "regex",
          "title": "local subnets (IANA reserved address space). Pattern exists because 'Do not use this proxy for localhost and intranet/private IP addresses' is checked.",
          "pattern": "^[^:]+://(?:[^:@/]+(?::[^@/]+)?@)?(?:192\\.168\\.\\d+\\.\\d+|10\\.\\d+\\.\\d+\\.\\d+|172\\.(?:1[6789]|2[0-9]|3[01])\\.\\d+\\.\\d+)(?::\\d+)?(?:/.*)?/",
          "active": true
        },
        {
          "type": "regex",
          "title": "localhost - matches the local host optionally prefixed by a user:password authentication string and optionally suffixed by a port number. The entire local subnet (127.0.0.0/8) matches. Pattern exists because 'Do not use this proxy for localhost and intranet/private IP addresses' is checked.",
          "pattern": "^[^:]+://(?:[^:@/]+(?::[^@/]+)?@)?[\\w-]+(?::\\d+)?(?:/.*)?/",
          "active": true
        }
      ]
    },
      ],
    }
  ],
  "showPatternProxy": false
}`

**Comment by erosman** [comment]
Although it should not cause any problems, the settings seems to be imported from older versions of FoxyProxy (pre v8) and uses older type patterns.
It might be useful to update those.

As mentioned in the included Help...

> Wildcard
> Leading or trailing `*` is not necessary

`*whatismyip*` would be the same as `whatismyip` (which is better)

AFA excludes ...
- if they are Global (for all proxies), then it would be more efficient to use Global Exclude
- If they are not global, then using the v8  ready=made patterns (via Quick Add) should be more efficient and easier to maintain

**Comment by dromhost** [comment]
Thanks 
Good clean suggestions on patterns. Was not aware of the change. Implemented and working well now.

Global Exclude: You mean something like bleow

**Comment by erosman** [comment]
> Global Exclude: You mean something like below?

Global Exclude is found on the Options Tab in FoxyProxy settings.
The included Help has information on how to use it.

**Comment by erosman** [comment]
Are you still experiencing the issue?

**Comment by dromhost** [comment]
Yes. Nothing has changed.  When I restart Firefox, patterns are not recognized even though "Proxy by Patterns" is checked. When I click "Disable" and then click "Proxy by Patterns", the SOCKS proxies start working.

I don't really know where to start looking for the reason. 

133.0.3 (64-bit) Mozilla Firefox Linux
FoxyProxy 8.9

**Comment by erosman** [comment]
I am working on the next release (v8.10).
If you can find out the cause, I will work on it.

I noticed that you have sync enabled.
Is it possible that it is syncing with another Firefox?

Try the following:
- Back up your settings
- Disable sync
- Restore Defaults
- Start fresh by adding one proxy with its patterns manually (no copy/paste from a previous versions)
- Test the result

Let me know of the outcome.
